cranelift-wasm/src/sections_translator.rs

```diff
@@ -338,3 +338,45 @@ pub fn parse_data_section<'data>(
 
     Ok(())
 }
+
+/// Parses the Name section of the wasm module.
+pub fn parse_name_section<'data>(
+    names: NameSectionReader<'data>,
+    environ: &mut dyn ModuleEnvironment<'data>,
+) -> WasmResult<()> {
+    for subsection in names {
+        match subsection? {
+            wasmparser::Name::Module(module_name) => {
+                let module_name = module_name.get_name()?;
+                environ.declare_module_name(module_name);
+            }
+            wasmparser::Name::Function(function_names) => {
+                let mut naming_reader = function_names.get_map()?;
+                for _ in 0..naming_reader.get_count() {
+                    let Naming { index, name } = naming_reader.read()?;
+                    environ.declare_func_name(FuncIndex::from_u32(index), name);
+                }
+            }
+            wasmparser::Name::Local(local_names) => {
+                let mut func_local_reader = local_names.get_function_local_reader()?;
+                for _ in 0..func_local_reader.get_count() {
+                    let func_local_names = func_local_reader.read()?;
+                    let func_index = func_local_names.func_index;
+                    let mut naming_reader = func_local_names.get_map()?;
+                    for _ in 0..naming_reader.get_count() {
+                        let Naming {
+                            index: local_index,
+                            name,
+                        } = naming_reader.read()?;
+                        environ.declare_local_name(
+                            FuncIndex::from_u32(func_index),
+                            LocalIndex::from_u32(local_index),
+                            name,
+                        );
```

Comment:
Local names are handled in this branch now, and the module name is handled above. Listing these variants' respective `match` arms in ascending order according to their subsection id's in the [wasm spec](https://webassembly.github.io/spec/core/appendix/custom.html#subsections).